src/main/java/org/gridsuite/network/map/NetworkMapService.java

```diff
@@ -109,6 +107,18 @@ public List<ElementInfos> getVoltageLevelBusesOrBusbarSections(UUID networkUuid,
         };
     }
 
+    public List<ElementInfosWithSwitchStatus> getVoltageLevelSwitches(UUID networkUuid, String voltageLevelId, String variantId) {
+        Network network = getNetwork(networkUuid, PreloadingStrategy.NONE, variantId);
+
+        return StreamSupport.stream(
+                        network.getVoltageLevel(voltageLevelId).getSwitches().spliterator(), false)
+                .map(sw -> ElementInfosWithSwitchStatus.builder()
+                        .id(sw.getId())
+                        .open(sw.isOpen())
+                        .build())
+                .collect(Collectors.toList());
```

Comment:
```suggestion
        List<ElementInfosWithSwitchStatus> switchInfosList = new ArrayList<>();
        network.getVoltageLevel(voltageLevelId).getSwitches().forEach(sw ->
                switchInfosList.add(ElementInfosWithSwitchStatus.builder()
                    .id(sw.getId())
                    .open(sw.isOpen())
                    .build()));
        return switchInfosList;
```
IMO, same number of lines and easy to understand, suitable for simple map without chainning operators